staging/src/k8s.io/code-generator/cmd/validation-gen/targets.go

```diff
@@ -186,21 +204,28 @@ func GetTargets(context *generator.Context, args *Args) []generator.Target {
 		}
 
 		// Find the right input pkg, which might not be this one.
-		inputPath := pkgToInput[i]
+		inputPath := pkgToInput[input]
 		typesPkg = context.Universe[inputPath]
 
+		// Figure out which types we should be considering further.
 		var rootTypes []*types.Type
 		for _, t := range typesPkg.Types {
 			if shouldCreateObjectValidationFn(t) {
 				rootTypes = append(rootTypes, t)
 			}
 		}
+		// Deterministic ordering helps in logs and debugging.
+		slices.SortFunc(rootTypes, func(a, b *types.Type) int {
+			return cmp.Compare(a.Name.String(), b.Name.String())
+		})
```

Comment:
Thank you.  I keep getting hit with "the generated code change because you didn't sort something" type issues.